onnxruntime/core/providers/mkldnn/math/sum.cc

```diff
@@ -0,0 +1,232 @@
+// Copyright(C) 2018 Intel Corporation
+// Licensed under the MIT License
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/math/sum.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN Sum primitives.
+struct SumParams {
+  std::vector<mkldnn::memory::dims> src_dims;
+  mkldnn::memory::dims& dst_dim;
+  int num_inputs;
+  int num_dimensions;
+
+  SumParams(std::vector<mkldnn::memory::dims> dims,
+    mkldnn::memory::dims& dst_dims, int &numinputs, int &dimensions)
+    : src_dims(dims),
+    dst_dim(dst_dims),
+    num_inputs(numinputs),
+    num_dimensions(dimensions) {}
+
+  // Used as the key for Sum Primitive Reuse Sum.
+  std::string ToString() const {
+    std::string key;
+    key.reserve(64);
+    key.append("sum_");
+    AddDimsToKey(key, src_dims[0]);
+    AddDimsToKey(key, dst_dim);
+    return key;
+  }
+};
+
+template <typename T>
+class SumPrimitive final : public PrimitiveBase {
+ public:
+  explicit SumPrimitive(const SumParams& params)
+      : cpu_engine_(GetEngine()) {
+    context_.stream.reset(new mkldnn::stream(mkldnn::stream::kind::eager));
+    if (context_.sum_pd == nullptr) {
+      Initialize(params);
+    }
+  }
+
+  ~SumPrimitive() = default;
+
+  void Compute(OpKernelContext* context, int& numinputs) {
+    const Tensor* X1 = context->Input<Tensor>(0);
+    Tensor* Y = context->Output(0, X1->Shape());
+    T* dst_data = Y->template MutableData<T>();
+
+    context_.dst_mem->set_data_handle(
+      static_cast<void*>(const_cast<T*>(dst_data)));
+
+    for (int i = 0; i < numinputs; i++) {
+      const Tensor* X = context->Input<Tensor>(i);
+      const T* src_data = X->template Data<T>();
+      context_.srcs_memory[i].set_data_handle(
+        static_cast<void*>(const_cast<T*>(src_data)));
+    }
+
+    context_.stream->submit(context_.net);
+  }
+
+  std::unique_ptr<mkldnn::memory::desc> GetDstMemoryDesc() 
+    const { return context_.dst_md; }
```

Comment:
const should be on the same line as the function

Reply:
done.